test/test_rpc_endpoints.py

```diff
@@ -0,0 +1,641 @@
+"""
+Tests RPC endpoints.
+"""
+from __future__ import annotations
+
+import json
+
+import pytest
+from starkware.starknet.public.abi import get_storage_var_address, get_selector_from_name
+
+from starknet_devnet.server import app
+
+from .util import (
+    load_file_content,
+)
+from .test_endpoints import send_transaction
+
+
+DEPLOY_CONTENT = load_file_content("deploy.json")
+INVOKE_CONTENT = load_file_content("invoke.json")
+
+
+def rpc_call(method: str, params: dict | list) -> dict:
+    """
+    Make a call to the RPC endpoint
+    """
+    req = {
+        "jsonrpc": "2.0",
+        "method": method,
+        "params": params,
+        "id": 0
+    }
+
+    resp = app.test_client().post(
+        "/rpc",
+        content_type="application/json",
+        data=json.dumps(req)
+    )
+    result = json.loads(resp.data.decode("utf-8"))
+    return result
+
+
+def gateway_call(method: str, **kwargs):
+    """
+    Make a call to the gateway
+    """
+    resp = app.test_client().get(
+        f"/feeder_gateway/{method}?{''.join(f'{key}={value}&' for key, value in kwargs.items()).rstrip('&')}"
+    )
+    return json.loads(resp.data.decode("utf-8"))
+
+
+@pytest.fixture(name="deploy_info")
+def fixture_deploy_info() -> dict:
+    """
+    Deploy a contract on devnet and return deployment info dict
+    """
+    resp = send_transaction(json.loads(DEPLOY_CONTENT))
+    deploy_info = json.loads(resp.data.decode("utf-8"))
+    return deploy_info
+
+
+@pytest.fixture(name="invoke_info")
+def fixture_invoke_info() -> dict:
+    """
+    Make a invoke transaction on devnet and return invoke info dict
+    """
+    invoke_tx = json.loads(INVOKE_CONTENT)
+    invoke_tx["calldata"] = ["0"]
+    resp = send_transaction(invoke_tx)
+    invoke_info = json.loads(resp.data.decode("utf-8"))
+    return invoke_info
+
+
+def pad_zero(felt: str) -> str:
+    """
+    Convert felt with format `0xValue` to format `0x0Value`
+    """
+    felt = felt.lstrip("0x")
+    return "0x0" + felt
+
+
+# pylint: disable=unused-argument
```

Comment:
Why suppresion? Can't you just not have the deploy_info parameter?

Reply:
Maybe `deploy_info` is not a good name, or a different fixture structure should be used instead?

Deploy_info fixture makes sure that there's always a block with some transaction before any of these tests is run. The problem was when i was running tests like `get_test_storage_at` on its own, it would fail because it tried to do `gateway_call("get_block", blockNumber=0)` even though there was no blocks.  

Perhaps this fixture should be changed to `@pytest.fixture(autouse=True)`?

Reply:
I'm referring to the pylint suppresion. It can be avoided by simply making function `test_get_block_by_number` not accept any parameters (since it doesn't use any, not even `deploy_info`).